docs/content/commands.md

```diff
@@ -70,34 +72,37 @@ public class BuildSystem : ComponentSystem
     {
         for(var i = 0; i < data.Length; i++)
         {
-            var requestSender = data.BuilderRequestSender[i];
+            var requestSender = data.BuildWallSender[i];
             var entityId = data.EntityIds[i];
             
-            BuildRequest buildRequest = new BuildRequest
+            Builder.BuildWall.Request request = new Builder.BuildWall.Request
             {
-                Location = new Location(...),
-                Rotation = new Rotation(...)
-            }
+                TargetEntityId = entityId,
+                Payload = new BuildRequest
+                {
+                    Location = new Location(...),
+                    Rotation = new Rotation(...)
+                }
+            };
+
             
-            requestSender.SendBuildWallRequest(entityId, buildRequest);
+            requestSender.RequestsToSend.Add(request);
         }
     }
 }
 ```
 
 This system runs on a client. It injects all SpatialOS entities in the client's view that have the `Builder` SpatialOS component, and sends a `build_wall` command to each SpatialOS entity (which will be received by the managed worker).
 
-To send a `build_wall` command in a system, you need to inject `CommandRequestSender<Builder>` into the system like any other ECS component.
+To send a `build_wall` command in a system, you need to inject `Builder.CommandSenders.BuildWall` into the system like any other ECS component.
 
 ### Responding to command requests
 
 When a worker instance receives a command request, the command request is represented with reactive ECS components. 
 
-The Unity GDK attaches a `CommandRequests<T>` ECS component to the specified ECS entity (where `T` implements `IIncomingCommandRequest`). `CommandRequests<T>` holds a list of `T`. The Unity GDK cleans it up at the end of the tick.
-
-You can use the object in the list itself to respond to that particular request.
+The Unity GDK attaches a `ComponentName.CommandRequests.CommandName` ECS component to the specified ECS entity, where `ComponentName` is the SpatialOS component the command is defined in and `CommandName` is the name of the command in schema. `ComponentName.CommandRequests.CommandName` contains a list of type `ComponentName.CommandName.ReceivedRequest`. The Unity GDK cleans it up at the end of the tick.
 
-> **Note**: `CommandRequests<T>` is a `Component`, not an `IComponentData`. This means you must use a `ComponentArray<T>` for injection rather than a `ComponentDataArray<T>`.
+To respond to the request, use `ComponentName.CommandResponders.CommandName` for that given command. This contains a list of type `ComponentName.CommandName.Response`. Create and add a `ComponentName.CommandName.Response` object to this list and the GDK will automatically send the response for you.
 
```

Comment:
No need to say "will automatically" - it's stronger (and suggests a trustworthy, reliable product) to write, "...and the GDK automatically sends the response for you.